R/bs-theme-preset.R

```diff
@@ -0,0 +1,80 @@
+new_theme_preset <- function(
```

Comment:
There are no other presets, so no need to quantify that it is `new`
```suggestion
theme_preset <- function(
```

Reply:
There are no global setter methods. If there were, they'd be `theme_preset_global()` (or similar).

Following pattern of `bs_theme()`, not `new_bs_theme()`...

Reply:
The distinction is that `bs_theme()` provides a whole sass bundle/theme, so `bs_theme_preset()` wouldn't be a good fit. There are other `theme_*()` functions that take a `theme` object and return a value, e.g. `theme_bootswatch()` takes `theme` and returns the name of the bootswatch theme. So the namespace is kinda crowded.

In this case, we're creating a light-weight S3 class, so I was following the naming recommendation in [Advanced R](https://adv-r.hadley.nz/s3.html#s3-classes):

> To avoid foot-bullet intersections when creating your own class, I recommend that you usually provide three functions:
>
> A low-level constructor, `new_myclass()`, that efficiently creates new objects with the correct structure.

Reply:
I did go through some name changes near the end of the PR, so maybe it should be `new_bs_preset()`.

Reply:
This method feels like a **user friendly** way to make the object. So it should be `theme_preset()`. (sans name clashing)

Your `as_preset()` is closer to the _light weight_ method that constructs the class object (ex: `new_preset_theme()`)

How about `preset_theme()` to get away from `theme_*()` name clashing?

Reply:
I made `new_bs_preset()` the _actual_ light-weight S3 class constructor and renamed `new_theme_preset()` to `resolve_bs_preset()`. This resulting in a much cleaner separation of concerns.